### PR TITLE
Avoid object cast

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

File: tests/tpch/dask_queries.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def query_2(dataset_path, fs, scale):
     )
     part_filtered = part_filtered[
         (part_filtered["p_size"] == var1)
-        & (part_filtered["p_type"].astype(str).str.endswith(var2))
+        & (part_filtered["p_type"].astype("string[pyarrow]").str.endswith(var2))
```
Comment (fjetter) [inline tests/tpch/dask_queries.py:69]: why is this cast necessary in the first place?

image

```suggestion
        & (part_filtered["p_type"].str.endswith(var2))
```

File: tests/tpch/dask_queries.py [hunk 2]
```diff
@@ -1106,7 +1106,7 @@ def query_20(dataset_path, fs, scale):
     ]
     res_1 = (
         res_1.groupby(["l_partkey", "l_suppkey"])["l_quantity"]
-        .sum()
+        .sum(split_out=True)
```
Comment (fjetter) [inline tests/tpch/dask_queries.py:1109]: Is this something we could do automatically? Why is this necessary?

File: tests/tpch/dask_queries.py (hunk 1 above)
Comment (phofl) [reply tests/tpch/dask_queries.py:69]: Very good point, I was just passing by here. The cast doesn't seem necessary

File: tests/tpch/dask_queries.py (hunk 2 above)
Comment (phofl) [reply tests/tpch/dask_queries.py:1109]: We are already shuffling but partitions are still too big after the groupby, which slows us down quite a bit. We might want to reevaluate the factor, but this doesn't really aggregate, so True is the best value here

Comment (phofl) [reply tests/tpch/dask_queries.py:1109]: Yeah there is always the case where we don't reduce much at all, we could use split_out=True by default but this creates also tiny partitions in other scenarios, I don't think that we can hit every case without cardinality information

File: tests/tpch/dask_queries.py [hunk 3]
```diff
@@ -1108,7 +1107,7 @@ def query_20(dataset_path, fs, scale):
     ]
     res_1 = (
         res_1.groupby(["l_suppkey", "l_partkey"])["l_quantity"]
-        .sum()
+        .sum(split_out=True)
```
Comment (phofl) [inline tests/tpch/dask_queries.py:1110]:
```suggestion
        .sum()
```

File: tests/tpch/dask_queries.py (hunk 2 above)
Comment (phofl) [reply tests/tpch/dask_queries.py]: nope this can go away

**Review by fjetter — APPROVED** [review]
